File: Source/Back-end/src/main/java/org/elis/manutenzione/dto/response/RifiutaManutenzioneResponse.java
package org.elis.manutenzione.dto.response;

import lombok.Data;

/**
 * DTO che rappresenta la risposta alla richiesta di rifiuto di una manutenzione.
 */
@Data
public class RifiutaManutenzioneResponse {

    /**
     * L'ID della manutenzione rifiutata.
     */
    private long id;

    /**
     * La descrizione della manutenzione.
     */
    private String descrizione;

    /**
     * Il nome del luogo in cui si sarebbe dovuta effettuare la manutenzione.
     */
    private String luogo;

    /**
     * Il nucleo a cui appartiene il luogo.
     */
    private String nucleo;

    /**
     * La priorità della manutenzione.
     */
    private int priorita;
}

File: Source/Front-end/front-end/src/app/model/response/rifiuta-manutenzione-response.ts
export class RifiutaManutenzioneResponse {

  id: number;
  descrizione: string;
  luogo: string;
  nucleo: string;
  priorita: number;

  constructor(id: number, descrizione: string, luogo: string, nucleo: string, priorita: number) {
    this.id = id;
    this.descrizione = descrizione;
    this.luogo = luogo;
    this.nucleo = nucleo;
    this.priorita = priorita;
  }

}
